src/lib/types.hpp

```diff
@@ -151,11 +151,12 @@ constexpr ValueID NULL_VALUE_ID{std::numeric_limits<ValueID::base_type>::max()};
 
 constexpr ValueID INVALID_VALUE_ID{std::numeric_limits<ValueID::base_type>::max()};
 
-// The Scheduler currently supports just these 2 priorities, subject to change.
+// The Scheduler currently supports just these 3 priorities, subject to change.
 enum class SchedulePriority {
-  Unstealable = 2,  // Schedule task at the end of the queue with disabled workstealing
-  Normal = 1,       // Schedule task at the end of the queue
-  High = 0          // Schedule task at the beginning of the queue
+  All = 3,     // Default priority when it comes to pulling tasks from the TaskQueue
+  Normal = 2,  // Schedule task at the end of the queue
```

Comment:
"Normal" sounds like "default" to me. Could you distinguish those two? Do we even need them?

Reply:
"Normal" is the "default" priority when scheduling. I added "All" only as a way for workers telling the TaskQueue, what priority they are able to pull:
```
task_queue.hpp
/* *
   * Returns a Tasks that is ready to be executed and removes it from the queue
   *
   * @param min_priority defines the minimum priority a tasks must hast to be pulled
   */
  std::shared_ptr<AbstractTask> pull(SchedulePriority min_priority = SchedulePriority::All);
```
Workers are also constructed with a minimum priority of tasks they can work on (so that the dedicated `JobTask` worker only works on `SchedulePriority::JobTask`. For readability I added this this `All` priority, that indicates there is no limit on what tasks can be worked on.

In theory I can remove `SchedulePriority::All` completely, and just have `SchedulePriority::Normal` as "All", or rename that to `SchedulePriority::Default`.

Reply:
Maybe `SchedulePriority min_priority = SchedulePriority::Lowest`? For now, we could have Lowest, Default, Highest.

Reply:
That sounds good. But we would have Lowest, Default, Highest, JobTask, right?

Reply:
Yes